.github/workflows/qns.yml

```diff
@@ -53,4 +52,21 @@ jobs:
             RUST_VERSION=stable
           cache-from: type=gha
           cache-to: type=gha,mode=max
-          platforms: linux/amd64, linux/arm64
+          # On pull requests only build amd64 for the sake of CI time.
+          platforms: ${{ github.event_name == 'pull_request' && 'linux/amd64' || 'linux/amd64, linux/arm64' }}
+          load: ${{ github.event_name == 'pull_request' }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Run QUIC Interop tests
+        if: ${{ github.event_name == 'pull_request' }}
+        # TODO: Replace once https://github.com/quic-interop/quic-interop-runner/pull/356 is merged.
+        uses: ./.github/actions/quic-interop-runner
+        with:
+          name: 'neqo-latest'
+          image: ${{ steps.docker_build_and_push.outputs.imageID }}
+          url: https://github.com/mozilla/neqo
+          test: handshake
```

Comment:
Testcase `handshake` for now to keep CI time low. I suggest we expand gradually once we gained some experience running the QUIC Interop tests on each pull request. Happy to change.